doc/source/whatsnew/v1.2.0.rst

```diff
@@ -309,6 +309,13 @@ Optional libraries below the lowest tested version may still work, but are not c
 
 See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for more.
 
+.. _whatsnew_200.api.other:
+
+Other API changes
+^^^^^^^^^^^^^^^^^
+
+- Sorting in descending order is now stable for :meth:`Series.sort_values` and :meth:`Index.sort_values` for DateTime-like :class:`Index` subclasses. This will affect sort order when sorting :class:`DataFrame` on multiple columns, sorting with a key function that produces duplicates, or requesting the sorting index when using :meth:`Index.sort_values`. When using :meth:`Series.value_counts`, count of missing values is no longer the last in the list of duplicate counts, and its position corresponds to the position in the original :class:`Series`. When using :meth:`Index.sort_values` for DateTime-like :class:`Index` subclasses, NaTs ignored the ``na_position`` argument and were sorted to the beggining. Now they respect ``na_position``, the default being ``last``, same as other :class:`Index` subclasses. (:issue:`35992`)
```

Comment:
Hey @AlexKirko, is it possible this is not only changed for datetime-like Index subclasses?

I see on released pandas this:

```
In [2]: pd.Index([0, 1, 0, 1]).value_counts()
Out[2]: 
1    2
0    2
dtype: int64
```

while on master / 1.2.0rc:

```
In [2]: pd.Index([0, 1, 0, 1]).value_counts()
Out[2]: 
0    2
1    2
dtype: int64
```

Reply:
Of course, it's the resulting Series that is sorted, not the Index. And I suppose for Series it's for all types?

(this change turned out to uncover a bug in statsmodels, https://github.com/statsmodels/statsmodels/pull/7215)